core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterCreationSetupService.java

```diff
@@ -322,7 +324,7 @@ private ClusterComponent determineHDPRepoConfig(Blueprint blueprint, long stackI
                 StackRepoDetails repo = createStackRepoDetails(stackRepoDetails, osType);
                 Optional<String> vdfUrl = getVDFUrlByOsType(osType, stackRepoDetails);
                 vdfUrl.ifPresent(s -> stackRepoDetails.getStack().put(CUSTOM_VDF_REPO_KEY, s));
-                if (ambariStackDetails != null) {
+                if (ambariStackDetails != null && ambariStackDetails.getMpacks() != null) {
```

Comment:
why?

Reply:
because I've encountered an issue when this mpack was null and in the body of this if statement we're processing it, so null check is necessary